File: Binding.Web/Models/Block.cs
using System;

namespace Binding.Models
{
    public class Block
    {
        public Guid Id { get; set; }
        public BlockType Type { get; set; }
        public string Content { get; set; }
        
        public DateTime Created { get; set; }
        public DateTime Updated { get; set; }
        
        public int Order { get; set; }
        
        // one to many pages
        public Page Page { get; set; }
    }

    public class BlockViewModel
    {
        
    }
}

File: Binding.Web/ClientApp/src/app/_models/Block.ts
import {BlockType} from "./BlockType";
import {Page} from "./Page";

export interface Block
{
  Id?: string;
  Type: BlockType;
  Content: string;
  Created?: Date;
  Updated?: Date;
  Order?: number;
  Page?: Page;
}

export interface BlockViewModel
{
  Id: string;
  Type: BlockType;
  Content: string;
  Created: Date;
  Updated: Date;
  Order: number;
}
